platform/darwin/src/NSComparisonPredicate+MGLAdditions.mm

```diff
@@ -318,14 +318,32 @@ - (id)mgl_jsonExpressionObject {
         case NSNotEqualToPredicateOperatorType:
             op = @"!=";
             break;
+        case NSBetweenPredicateOperatorType: {
+            op = @"all";
+            NSArray *limits = self.rightExpression.constantValue;
+            NSPredicate *leftHandPredicate = [NSComparisonPredicate predicateWithLeftExpression:limits[0]
+                                                                                rightExpression:self.leftExpression
+                                                                                       modifier:NSAllPredicateModifier
+                                                                                           type:NSLessThanOrEqualToPredicateOperatorType
```

Comment:
Oh, I meant that we could translate `a BETWEEN {b, c}` to `a >= min({b, c}) AND a <= max({b, c})` then convert to JSON. I don’t see why that wouldn’t work, but it certainly would complicate any attempt to round-trip back to NSPredicate, if we decide later on that that would be desirable.

Reply:
What I meant is if you evaluate the predicate `a >= min({3, 1}) AND a <= max({3, 1})` to ensure that `BETWEEN` is in the correct order it won't work since evaluating to a `keyPath a` does not contain any value at that moment. If we evaluate a constant value like  `2 >= min({3, 1}) AND 2 <= max({3, 1})` then it will.

Reply:
Couldn’t we do something like this?

```objc
[NSPredicate predicateWithFormat:@"%@ >= min(%@) AND %@ <= max(%@)", self.leftExpression, self.rightExpression, self.leftExpression, self.rightExpression];
```